File: Searching an element in a sorted array - GFG/searching-an-element-in-a-sorted-array.c
// { Driver Code Starts
//Initial Template for C

#include <stdio.h> 


 // } Driver Code Ends
//User function Template for C

int searchInSorted(int arr[], int N, int K) 
{ 
    
       // Your code here
       int low=0,high=N-1;
       while(low<=high)
       {
           int mid=(low+high)/2;
           if(arr[mid]<K)
               low=mid+1;
           else if(arr[mid]>K)
               high=mid-1;
           else
               return 1;
       }
       return -1;
}

// { Driver Code Starts.


int main(void) 
{ 
    
    int t;
    scanf("%d", &t);
    while(t--){
        int n, k;
        scanf("%d%d", &n, &k);
        
        int arr[n];
        
        for(int i = 0;i<n;i++){
            scanf("%d", &arr[i]);
        }
        
        printf("%d\n", searchInSorted(arr, n, k));

    }

	return 0; 
} 
  // } Driver Code Ends

File: Searching an element in a sorted array - GFG/searching-an-element-in-a-sorted-array.java
// { Driver Code Starts
import java.util.*;
import java.lang.*;
import java.io.*;

class gfg
{
    
    public static void main(String args[])throws IOException
    {
        BufferedReader read = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        
        while(t-- > 0)
        {
            String s[] = read.readLine().trim().split("\\s+");
            
            int n = Integer.parseInt(s[0]);
            int k = Integer.parseInt(s[1]);
            
            int arr[] = new int[n];
            
            String st[] = read.readLine().trim().split("\\s+");
            
            for(int i = 0; i < n; i++)
            {
                arr[i] = Integer.parseInt(st[i]);
            }
            
            Solution obj = new Solution();
            
            System.out.println(obj.searchInSorted(arr, n, k));
        }
    }
}// } Driver Code Ends


class Solution{
    static int searchInSorted(int arr[], int N, int K)
    {
        
        // Your code here
        int low=0,high=N-1;
       while(low<=high)
       {
           int mid=(low+high)/2;
           if(arr[mid]<K)
               low=mid+1;
           else if(arr[mid]>K)
               high=mid-1;
           else
               return 1;
       }
       return -1;
        
    }
}